packages/manager/src/features/CloudPulse/Widget/components/CloudPulseIntervalSelect.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+
+import { Autocomplete } from 'src/components/Autocomplete/Autocomplete';
+
+import type { TimeGranularity } from '@linode/api-v4';
+
+export interface IntervalSelectProperties {
+  /**
+   * Default time granularity to be selected
+   */
+  default_interval?: TimeGranularity | undefined;
+
+  /**
+   * Function to be triggered on aggregate function changed from dropdown
+   */
+  onIntervalChange: any;
+
+  /**
+   * scrape intervalto filter out minimum time granularity
+   */
+  scrape_interval: string;
+}
+
+export const getInSeconds = (interval: string) => {
+  if (interval.endsWith('s')) {
+    return Number(interval.slice(0, -1));
+  } else if (interval.endsWith('m')) {
+    return Number(interval.slice(0, -1)) * 60;
+  } else if (interval.endsWith('h')) {
+    return Number(interval.slice(0, -1)) * 3600;
+  } else if (interval.endsWith('d')) {
+    return Number(interval.slice(0, -1)) * 86400;
+  }
```

Comment:
Since we're returning early we can unnest these ifs

```suggestion
  if (interval.endsWith('s')) {
    return Number(interval.slice(0, -1));
  }
  if (interval.endsWith('m')) {
    return Number(interval.slice(0, -1)) * 60;
  }
  if (interval.endsWith('h')) {
    return Number(interval.slice(0, -1)) * 3600;
  }
  if (interval.endsWith('d')) {
    return Number(interval.slice(0, -1)) * 86400;
  }
```

Reply:
updated, thanks